staging/src/k8s.io/apimachinery/pkg/runtime/helper.go

```diff
@@ -236,6 +236,19 @@ func (e WithVersionEncoder) Encode(obj Object, stream io.Writer) error {
 			gvk = preferredGVK
 		}
 	}
+	if gvk == oldGVK {
+		// gvk already set as intended, skip injecting it.
+		return e.Encoder.Encode(obj, stream)
```

Comment:
Might be worth merging also without the other changes?

Reply:
Yes please

Reply:
=> https://github.com/kubernetes/kubernetes/pull/121771